### PR TITLE
[export] [cc] Support ti.RecordGroupHint to group launched kernels

### PR DESCRIPTION
**Pull request by archibate**
Related issue = [数据删除]

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
I need this for exporting `mpm_lagrangian_forces`.

### CODE REVIEW

**Comment by archibate** [comment]
@yuanming-hu I will merge this in 24 hours if you don't reject me, otherwise our work in TaichiActor will be blocked, thank for understanding.

**Comment by Hanke98** [comment]
> @yuanming-hu I will merge this in 24 hours if you don't reject me, otherwise our work in TaichiActor will be blocked, thank for understanding.

I am so sorry that I didn't see your request until today, I will review your code asap.

File: python/taichi/core/record.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        start_recording(self.filename)
+        return self
+
+    def __exit__(self, *args):
+        stop_recording()
+
+
+class RecordGroupHint:
+    def __init__(self, name):
+        if name in recorded:
```
Comment (Hanke98) [inline python/taichi/core/record.py:33]: I am not sure if this is correct. Maybe you should use `RecordGroupHint.record` or `self.record` here.

**Review by Hanke98 — COMMENTED** [review]
Cool! I reviewed your code and I thought it is very very useful to record the launch order in group.

However, when I tried to run `demo_record_group_hint.py` I got:

[I 09/21/20 19:45:31.023] [action_recorder.cpp:start_recording@27] ActionRecorder: start recording to [record.yml]
[W 09/21/20 19:45:31.023] [__init__.py:adaptive_arch_select@552] Arch=[Arch.cc] is not supported, falling back to CPU
[Taichi] Starting on arch=x64
Traceback (most recent call last):
  File "demo_record_group_hint.py", line 22, in <module>
    with ti.RecordGroupHint('my_substep'):
  File "/Users/liujiafeng/playground/projects/taichi/python/taichi/core/record.py", line 33, in __init__
    if name in recorded:
NameError: name 'recorded' is not defined

I think the variable `recorded` in `RecordGroupHint` class may be the reason. Could you please help me to run this file correctly? Thanks a lot!

**Review by Hanke98 — APPROVED** [review]
Great! LGTM! Thanks for writing such useful tool! By the way, can we record kernels in group for other backends, such as metal?

**Pull request merged**